### DISCUSSION TITLE
Deprecate `first order` promotion rule for `nth order`

### DISCUSSION THREAD

**Discussion by seand7565** [solidusio/solidus#4410] [New Features or Ideas]
The [nth order](https://github.com/solidusio/solidus/blob/7d7b64da3860c55f2dfe0b229fa92a6086f7d5cf/core/app/models/spree/promotion/rules/nth_order.rb) promotion rule handles applying promotions to a users nth order. The only catch is, N can't be 1. The comment on the rule seems to indicate that the reason for this is that we already have a [first order](https://github.com/solidusio/solidus/blob/7d7b64da3860c55f2dfe0b229fa92a6086f7d5cf/core/app/models/spree/promotion/rules/first_order.rb)  promotion rule. But given that there's no real distinction between `first order` and `nth order where N=1`, and a user should expect to be able to set N to 1, I would expect `nth order` to fully encompass `first order`s functionality.

In the interest of keeping the promotions system as light as possible, we could deprecate the first_order promotion rule in favor of the nth order rule. We'd need to move a small amount of logic from `first_order` to `nth_order`, but it seems like a small enough change, and since that logic only applies to N=1 - which currently is impossible because of the validations on the rule - we won't be breaking `nth_order` for anyone.

**Comment by waiting-for-dev** [comment]
I think it makes total sense.
